common/core/nesteddisplay.cpp

```diff
@@ -22,13 +22,78 @@
 #include <sstream>
 
 #include <hwctrace.h>
+#include <hwclayer.h>
+#include "utils_android.h"
+#include <sys/socket.h>
+#include <netinet/in.h>
 
 namespace hwcomposer {
+#define METADATA_BUFFER_SIZE 12000
+#define METADATA_STREAM_START 0xF00D
+#define METADATA_STREAM_END 0xCAFE
+
+static char buf[METADATA_BUFFER_SIZE];
+
+#define HYPER_DMABUF_PATH "/dev/hyper_dmabuf"
+android::sp<SocketThread> NestedDisplay::st_ = NULL;
+int NestedDisplay::mclient_sock_fd = -1;
+
+NestedDisplay::NestedDisplay(NativeBufferHandler *buffer_handler) {
+  width_ = 1920;
+  height_ = 1080;
+
+  buffer_handler_ = buffer_handler;
+  struct ioctl_hyper_dmabuf_tx_ch_setup msg;
+  int ret;
+
+  mHyperDmaBuf_Fd = open(HYPER_DMABUF_PATH, O_RDWR);
+  if (mHyperDmaBuf_Fd < 0)
+    ALOGE("Hyper DmaBuf: open hyper dmabuf device node %s failed because %s", HYPER_DMABUF_PATH, strerror(errno));
+  else {
+    ALOGD("Hyper DmaBuf: open hyper dmabuf device node %s successfully!", HYPER_DMABUF_PATH);
+    /* TODO: add config option to specify which domains should be used, for now we share always with dom0 */
+    msg.remote_domain = 0;
+    ret = ioctl(mHyperDmaBuf_Fd, IOCTL_HYPER_DMABUF_TX_CH_SETUP, &msg);
```

Comment:
Please have this under compiler checks. Only have these IOCTLs need to be under checks.